### PR TITLE
e2e: api: add and use support to sort tolerations

### PR DESCRIPTION
**Pull request by ffromani**
Make sure we compare sorted tolerations in the e2e tests, so we minimize flakes.
No intended changes in behavior.

### CODE REVIEW

**Comment by ffromani** [comment]
related to #808

**Comment by ffromani** [comment]
requires: #888

File: test/e2e/serial/tests/tolerations.go [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+
+	nrtv1alpha2 "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/apis/topology/v1alpha2"
+	nropv1 "github.com/openshift-kni/numaresources-operator/api/numaresourcesoperator/v1"
```
Comment (shajmakh) [inline test/e2e/serial/tests/tolerations.go:32]: can be a little bit below with the locals as one group

Comment (ffromani) [reply test/e2e/serial/tests/tolerations.go]: fixed, thanks

**Pull request merged**